### ISSUE TITLE
An Analysis of Scale Invariance in Object Detection - SNIP

### ISSUE THREAD

**Issue by Swall0w** [Swall0w/papers#333]
Singh, Bharat, Davis, Larry S.

An analysis of different techniques for recognizing and detecting objects
under extreme scale variation is presented. Scale specific and scale invariant
design of detectors are compared by training them with different configurations
of input data. To examine if upsampling images is necessary for detecting small
objects, we evaluate the performance of different network architectures for
classifying small objects on ImageNet. Based on this analysis, we propose a
deep end-to-end trainable Image Pyramid Network for object detection which
operates on the same image scales during training and inference. Since small
and large objects are difficult to recognize at smaller and larger scales
respectively, we present a novel training scheme called Scale Normalization for
Image Pyramids (SNIP) which selectively back-propagates the gradients of object
instances of different sizes as a function of the image scale. On the COCO
dataset, our single model performance is 45.7% and an ensemble of 3 networks
obtains an mAP of 48.3%. We use ImageNet-1000 pre-trained models and only train
with bounding box supervision. Our submission won the Best Student Entry in the
COCO 2017 challenge. Code will be made available at this http URL

https://arxiv.org/abs/1711.08189
